### ISSUE TITLE
False positive: Use of moved value

### ISSUE THREAD

**Issue by Stzx** [intellij-rust/intellij-rust#5151]
## Environment

* **IntelliJ Rust plugin version:** 0.2.118.2171-193
* **Rust toolchain version:** cargo 1.44.0-nightly (7019b3ed3 2020-03-17)
* **IDE name and version:** IDEA 2019.3.4
* **Operating system:** Windows 10

## Problem description

Using let expression to deconstruct object, but tip error(`Use of moved value`)

But the compilation can pass.

## Steps to reproduce

Use the following code:
```rust
#[derive(Debug)]
struct Vector2f {
    x: f32,
    y: f32,
}

fn main() {
    let vector = Vector2f { x: 1.0, y: 3.0 };
    let new_vector = Vector2f { x: 2.0, ..vector };

    println!("{:?}", new_vector);

    let Vector2f { x: old_x, y: old_y } = vector;
    // Tips `Use of moved value`

    println!("{:?}", old_x);
    println!("{:?}", old_y);
}
```
